packages/fiori-elements-opa-writer/src/index.ts

```diff
@@ -3,12 +3,45 @@ import { create as createStorage } from 'mem-fs';
 import type { Editor } from 'mem-fs-editor';
 import { create } from 'mem-fs-editor';
 import type { ManifestNamespace } from '@sap-ux/ui5-config';
-import type { FEV4OPAConfig } from './types';
+import type { FEV4OPAConfig, FEV4OPAPageConfig } from './types';
+import { SupportedPageTypes } from './types';
 
 type Manifest = ManifestNamespace.SAPJSONSchemaForWebApplicationManifestFile & { [key: string]: any };
 
 /**
- * Create the configuration object from the app descriptor and the FLP app name.
+ * Create the page configuration object from the app descriptor and the target key.
+ * @param manifest - the app descriptor of the app
+ * @param targetKey - the key of the target in the manifest
+ * @returns Page configuration object, or undefined if the target type is not supported
+ */
+function createPageConfig(manifest: Manifest, targetKey: string): FEV4OPAPageConfig | undefined {
+    const appTargets = manifest['sap.ui5']?.routing?.targets;
+    const target = appTargets && appTargets[targetKey];
```

Comment:
Would it be possible to define the type for target somewhere central or at least here, so that you don't have to do it again and again below.

Reply:
Not sure I get what you mean...

Reply:
In line 19, you are getting a reference to `target`. The type that reference seems not to be clear because you have to explicitly define the type of target in line 26, 28 and 35. In each of the lines, you don't have any conditions leading to the type you used, therefore, I'd say that you already know in line 19 exactly the type of target (or at least you exactly know a subset).
So, line 19 should look like `const target: { name?: string; options?: OptionsTypeXYZ } = ...;` BUT if that is not allowed, then I am wondering if you are making an incorrect assumption of what `appTargets && appTargets[targetKey]` will be.